src/common/SubProcess.h

```diff
@@ -107,12 +116,21 @@ class SubProcessTimed : public SubProcess {
 		  std_fd_op stdout_op = CLOSE, std_fd_op stderr_op = CLOSE,
 		  int timeout = 0, int sigkill = SIGKILL);
 
+  #ifdef _WIN32
+  int spawn() override;
+  int join() override;
+  #endif
+
 protected:
   void exec() override;
 
 private:
   int timeout;
   int sigkill;
+
+  #ifdef _WIN32
+  std::thread *monitor;
```

Comment:
```suggestion
  std::thread waiter;
```

no need to use a pointer, please use a pain `std::thread` and check `monitor.get_id()` instead.